src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -342,61 +370,103 @@ private InitCodeLineView generateMapInitCodeLine(
+
+    InitValueConfig initValueConfig = item.getInitValueConfig();
+    Field field = item.getField();
+
+    if (context.getFeatureConfig().useResourceNameFormatOption(item.getField())
+        && !item.getType().isRepeated()) {
+      // For a repeated type, we want to use a SimpleInitValueView
+
+      ResourceNameInitValueView.Builder initValue = ResourceNameInitValueView.newBuilder();
+
+      initValue.resourceTypeName(ResourceNameUtil.getResourceName(field));
+
+      List<String> varList =
+          Lists.newArrayList(ResourceNameUtil.getResourceNamePathTemplate(field).vars());
+      initValue.formatArgs(getFormatFunctionArgs(varList, initValueConfig));
+
+      return initValue.build();
+    } else if (initValueConfig.hasFormattingConfig()) {
       FormattedInitValueView.Builder initValue = FormattedInitValueView.newBuilder();
 
       initValue.apiWrapperName(context.getNamer().getApiWrapperClassName(context.getInterface()));
       initValue.formatFunctionName(
           context.getNamer().getFormatFunctionName(initValueConfig.getCollectionConfig()));
-      List<String> formatFunctionArgs = new ArrayList<>();
-      for (String entityName : initValueConfig.getCollectionConfig().getNameTemplate().vars()) {
-        String entityValue =
-            "\"[" + LanguageUtil.lowerUnderscoreToUpperUnderscore(entityName) + "]\"";
-        if (initValueConfig.hasFormattingConfigInitialValues()
-            && initValueConfig.getCollectionValues().containsKey(entityName)) {
-          entityValue = initValueConfig.getCollectionValues().get(entityName);
-        }
-        formatFunctionArgs.add(entityValue);
-      }
-      initValue.formatArgs(formatFunctionArgs);
+
+      List<String> varList =
+          Lists.newArrayList(initValueConfig.getCollectionConfig().getNameTemplate().vars());
+      initValue.formatArgs(getFormatFunctionArgs(varList, initValueConfig));
 
       return initValue.build();
     } else {
       SimpleInitValueView.Builder initValue = SimpleInitValueView.newBuilder();
 
       if (initValueConfig.hasSimpleInitialValue()) {
         initValue.initialValue(
-            context.getTypeTable().renderPrimitiveValue(type, initValueConfig.getInitialValue()));
+            context
+                .getTypeTable()
+                .renderPrimitiveValue(item.getType(), initValueConfig.getInitialValue()));
       } else {
-        initValue.initialValue(context.getTypeTable().getZeroValueAndSaveNicknameFor(type));
+        initValue.initialValue(
+            context.getTypeTable().getZeroValueAndSaveNicknameFor(item.getType()));
       }
 
       return initValue.build();
     }
   }
 
+  private static List<String> getFormatFunctionArgs(
+      List<String> varList, InitValueConfig initValueConfig) {
+    List<String> formatFunctionArgs = new ArrayList<>();
+    for (String entityName : varList) {
+      String entityValue =
+          "\"[" + LanguageUtil.lowerUnderscoreToUpperUnderscore(entityName) + "]\"";
```

Comment:
I will use Name class on the entityName. Currently the Name class doesn't support special characters (like " and [). Should I add support for this, or do we want to keep the Name class only applicable to characters that are valid in normal names?